cases/api_views.py

```
@@ -477,10 +485,63 @@ def create(self, request, *args, **kwargs):
+            return self._get_object_by_court_case(lookup_value)
+        return super().get_object()
+
+    def _get_object_by_court_case(self, lookup_value):
+        court_identifier, _, raw_case_number = lookup_value.partition(":")
+        court_identifier = court_identifier.strip()
+        raw_case_number = raw_case_number.strip()
+
+        try:
+            case_number = normalize_case_number(raw_case_number)
+        except ValueError:
+            raise NotFound(f"No case found for court case '{lookup_value}'.")
```

Ruff (0.15.17)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)